src/main/java/frc/robot/Robot.java

```diff
@@ -44,6 +47,11 @@ public void disabledPeriodic() {}
 
     @Override
     public void autonomousInit() {
+        // disable vision
+        if (robotContainer.driveSubsystem instanceof BaseSwerveSubsystem) {
+            ((SwerveSubsystem) robotContainer.driveSubsystem).VISION_ENABLE = false;
```

Comment:
Cast to `BaseSwervesubsystem` instead? This will also make the shuffleboard vision enable entry not technically representative of the actual state of vision enabling, but that's lower priority; can be fixed by just making a method `setVisionEnable()` that changes both the boolean and the value of the entry.

Reply:
@ky28059 are you down to implement this?